### ISSUE TITLE
Unable to include } in a template (Generate Release Notes)

### ISSUE THREAD

**Issue by jynxeh** [rfennell/AzurePipelines#599]
### Azure DevOps Extension you are using
Generate Release Notes (Crossplatform)

### Where are you running it?
- Azure DevOps Service (VSTS)

### Version of Extension/Task
2.19.17

### Expected behaviour and actual behaviour
I'm attempting to use this extension to produce an HTML document however I'm struggling to work out which characters require escaping and which don't. Through trial and error I've worked out that " needs to be escaped to \" and { to \{ however haven't been able to find a way to escape }. I can produce 2 different error depending if I try to escape or not, both are listed below.

### Steps to reproduce the problem
Create a new step using this task and configure as normal. For the template use the below text:

this is a test }

When you run this step you will see the error:

OUTPUT:
Getting the current build details
Expanding the truncated commit messages...
Finished expanding [0] commits.
Get details of [0] WIs
Total commits: [0]
Total workitems: [0]
Leaving WI in default order as returned by API
Using in-line template
##[error]ReferenceError: Invalid left-hand side expression in postfix operation
##[error]ReferenceError: Invalid left-hand side expression in postfix operation

If you change the template to:

this is a test \}

OUTPUT:
Getting the current build details
Expanding the truncated commit messages...
Finished expanding [0] commits.
Get details of [0] WIs
Total commits: [0]
Total workitems: [0]
Leaving WI in default order as returned by API
Using in-line template
##[error]SyntaxError: Invalid or unexpected token
##[error]SyntaxError: Invalid or unexpected token

**Comment by rfennell** [comment]
The sample templates can be found [here](https://github.com/rfennell/AzurePipelines/tree/master/SampleTemplates/XplatGenerateReleaseNotes%20(Node%20based)/Version%202)

The way the task work is that it evaluates each line of the template as a line of node. So this means anything within a dollar and curly brackets e.g. `${foo}` is evaluated as code.

The basic objects you have access to are
- releaseDetails
- compareReleaseDetails
- buildetails (only inside the BUILDLOOP if running in a release) 
- widetail (only inside the WILOOP)
- csdetails (only inside the CSLOOP)

If you want to the properties any object use JSON.Stringify e.g.` ${JSON.stringify(releaseDetails)}`

Bottom line is I would avoid {} in your text

**Comment by rfennell** [comment]
As per my standard practice I am closing this issue as I have heard nothing for a while. If this issue remains a problem please do not hesitate to re-open it

**Comment by MitchDart** [comment]
I would like to know if we could possibly reopen this issue. I would like to use a JSON template because the release notes which I'm generating get posted off to Google chat using a JSON template so that I can structure the message in a card. 

For example: { title: "${releaseDetails.name......)} }

I have tried to use all kind of escapes for the {} but none of them seems to work.

**Comment by jynxeh** [comment]
As a sie note, I avoided this issue in my case by only generating the HTML table with this task, and then wrapping it with the HTML top and bottom (including CSS) when I published. This avoided the need to use curly brackets in this step. Appreciate this won't work you @MitchDart, although you could potentially build your JSON using regular brackets and replace them later. Not ideal I know, but might be a work around.
